sdk/core/azure-core/src/main/java/com/azure/core/util/paging/ContinuablePagedIterable.java

```diff
@@ -0,0 +1,147 @@
+ *
+ * @see IterableStream
+ * @see ContinuablePagedFlux
+ */
+public abstract class ContinuablePagedIterable<C, T, P extends ContinuablePage<C, T>> extends IterableStream<T> {
+    private final ContinuablePagedFlux<C, T, P> pagedFlux;
+    private final int batchSize;
+
+    /**
+     * Creates instance with the given {@link ContinuablePagedFlux}.
+     *
+     * @param pagedFlux the paged flux use as iterable
+     */
+    public ContinuablePagedIterable(ContinuablePagedFlux<C, T, P> pagedFlux) {
+        this(pagedFlux, 1);
+    }
+
+    /**
+     * Creates instance with the given {@link ContinuablePagedFlux}.
+     *
+     * @param pagedFlux the paged flux use as iterable
+     * @param batchSize the bounded capacity to prefetch from the {@link ContinuablePagedFlux}
+     */
+    public ContinuablePagedIterable(ContinuablePagedFlux<C, T, P> pagedFlux, int batchSize) {
+        super(pagedFlux);
+        this.pagedFlux = pagedFlux;
+        this.batchSize = batchSize;
+    }
+
+    /**
+     * Retrieve the {@link Stream}, one page at a time.
+     * It will provide same {@link Stream} of T values from starting if called multiple times.
+     *
+     * @return {@link Stream} of a pages
+     */
+    public Stream<P> streamByPage() {
+        return pagedFlux.byPage().toStream(this.batchSize);
+    }
+
+    /**
+     * Retrieve the {@link Stream}, one page at a time, starting from the next page associated with the given
+     * continuation token. To start from first page, use {@link #streamByPage()} instead.
+     *
+     * @param continuationToken The continuation token used to fetch the next page
+     * @return @return {@link Stream} of a pages
+     */
+    public Stream<P> streamByPage(C continuationToken) {
+        return this.pagedFlux.byPage(continuationToken).toStream(this.batchSize);
+    }
+
+    /**
+     * Retrieve the {@link Stream}, one page at a time, with each page containing {@code preferredPageSize}
+     * items.
+     *
+     * It will provide same {@link Stream} of T values from starting if called multiple times.
+     *
+     * @param preferredPageSize the preferred page size, service may or may not honor the page
+     *                          size preference hence client MUST be prepared to handle pages
+     *                          with different page size.
+     * @return @return {@link Stream} of a pages
+     */
+    public Stream<P> streamByPage(int preferredPageSize) {
+        return this.pagedFlux.byPage(null, preferredPageSize).toStream(this.batchSize);
+    }
+
+    /**
+     * Retrieve the {@link Stream}, one page at a time, with each page containing {@code preferredPageSize}
+     * items, starting from the next page associated with the given continuation token.
+     * To start from first page, use {@link #streamByPage()} or {@link #streamByPage(int)} instead.
+     *
+     * @param preferredPageSize the preferred page size, service may or may not honor the page
```

Comment:
Instead of referring to a service, we can just say "source may or may not honor the page size" as this paged iterable is a generic concept that can be applicable to non-service sources too.